barretenberg/bbup/bbup.ts

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+import { Command, Option } from "commander";
+const program = new Command();
+import { installBB } from "./shell.js";
+import ora from "ora";
+import logSymbols from "log-symbols";
+import { getBbVersionForNoir } from "./versions.js";
+import { execSync } from "child_process";
+
+const spinner = ora({ color: "blue", discardStdin: false });
+
+const bbup = program
+  .command("install", { isDefault: true })
+  .description("Installs Barretenberg.")
+  .addOption(
+    new Option(
+      "-v, --version <version>",
+      "The Barretenberg version to install"
+    ).implies({ noirVersion: null })
+  )
+  .addOption(
+    new Option(
+      "-nv, --noir-version <version>",
```

Comment:
```suggestion
      "-nv, --noir-version <noirVersion>",
```

(Is this needed?)

Reply:
mheee not so much now since we agreed supporting other frontends is not worth the hassle, and so we assume people always want to install the bb version compatible with their Nargo version

I will leave it for now, as there could be a scenario where someone wants to install a bb version for a different noir version than they have installed

Reply:
Oh I simply meant it for clarity / if it affects how the script works, not necessarily changing behaviors. All good if that's not a problem 🙌